cmake/CMakeLists.txt

```diff
@@ -84,8 +84,44 @@ function(setup_project NAME)
     endif()
 endfunction()
 
+function(find_executable VAR NAME BINARY VERSION_ARG)
+    if(DEFINED CACHE{${VAR}})
+        return()
+    endif()
+    find_program(${VAR} ${BINARY} ${ARGN})
+    if(NOT ${VAR})
+        return()
+    endif()
+    execute_process(COMMAND ${${VAR}} ${VERSION_ARG} OUTPUT_VARIABLE VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REGEX MATCH "(^|[ \n])([0-9][.0-9]*)([\n ]|$)" VERSION "${VERSION}")
+    message(STATUS "Found ${NAME}: ${${VAR}} (found version \"${CMAKE_MATCH_2}\")")
+endfunction()
+
 # }}}.
 
+# Git. {{{
+
+find_package(Git REQUIRED)
```

Comment:
This refers to the plain Git executable, not Git development files or libraries?

Reply:
Yes, this is just to get a trace with the git version on setup, and an early error if git is missing in some corner cases (building in a docker mounted volume).